From e682e81032bc823954942b385722ca40bf39e967 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 7 Jun 2019 10:12:29 +0200
Subject: [PATCH] x86/xen: irq_pipeline: disable Xen support if pipelining

The current IRQ_PIPELINE implementation does not cope with
paravirtualized hardware interrupt state management such as
PARAVIRT_XXL. When pipelining is enabled, disable Xen which is the
only in-tree user so far.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/x86/xen/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/xen/Kconfig b/arch/x86/xen/Kconfig
index ba5a41828e9d..20514f255127 100644
--- a/arch/x86/xen/Kconfig
+++ b/arch/x86/xen/Kconfig
@@ -5,7 +5,7 @@
 
 config XEN
 	bool "Xen guest support"
-	depends on PARAVIRT
+	depends on PARAVIRT && !IRQ_PIPELINE
 	select PARAVIRT_CLOCK
 	select X86_HV_CALLBACK_VECTOR
 	depends on X86_64 || (X86_32 && X86_PAE)
-- 
2.16.4

